dynosaur_derive/src/sig.rs

```diff
@@ -38,7 +38,7 @@ pub(crate) fn is_rpit(sig: &Signature) -> bool {
     match sig {
         Signature {
             asyncness: Some(_), ..
-        } => true,
+        } => false,
```

Comment:
This doesn't introduce any regression, as `is_rpit` is always checked after checking `is_async`. The new semantics of the function makes more sense, as this is true just when there's an explicit rpit showing up and `is_async` would check the asyncness and if the return type is impl Future.